libraries/standard/http/include/http_client.h

```diff
@@ -0,0 +1,339 @@
+   
+    /**
+     * @brief Flags to activate other request header configurations. 
+     */
+    uint32_t flags;
+} HTTPRequestInfo_t;
+
+/**
+ * @brief Callback to intercept headers during the first parse through of the 
+ * response as it is received from the network.
+ */
+typedef struct httpResponseParsingCallback {
+    /**
+     * Invoked when both a header field and its associated header value is found.
+     * @param[in] pContext User context.
+     * @param[in] fieldLoc Pointer to the first char of the header field name in the response buffer.
+     * @param[in] fieldLen Length in bytes of the field name.
+     * @param[in] valueLoc Pointer to the first char of the header value in the response buffer.
+     * @param[in] valueLen Length in bytes of the value.
+     * @param[in] statusCode The HTTP response status-code.
+     */
+    void (*onHeaderCallback)( void* pContext, const char* fieldLoc, size_t fieldLen, const char* valueLoc, size_t valueLen, uint16_t statusCode );
+
+    /* Private context for the application. */
+    void *pContext;
+} httpclientHeaderParsingCallback_t;
+
+typedef struct HTTPResponse
+{
+    /**
+     * @brief Buffer for both the raw HTTP header and body. 
+     * 
+     * This buffer cannot be modifed until #HTTPClient_Send returns.
+     */
+    uint8_t* pBuffer;
+    size_t bufferLen; /**< The length of the response buffer in bytes. */
+
+    /**
+     * @brief Optional callback for intercepting the header during the first 
+     * parse through of the response as is it receive from the network.
+     */
+    httpclientHeaderParsingCallback_t* pHeaderParsingCallback;
+    
+    /**
+     * @brief The Location of the response headers in pBuffer.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    uint8_t* pHeaders;
+
+    /**
+     * @brief Byte length of the response headers in pBuf.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    size_t headersLen; 
+
+    /**
+     * @brief The location of the response body in pBuf.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    uint8_t* pBody;
+
+    /**
+     * @brief Byte length of the body in pBuf.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    size_t bodyLen;
+    
+    /* Useful HTTP header values found. */
+
+    /**
+     * @brief The HTTP response Status-Code.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    uint16_t statusCode;
+    
+    /**
+     * @brief The value in the "Content-Length" header is returned here.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    size_t contentLength;
+
+    /**
+     * @brief Count of the headers sent by the server.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    size_t headerCount;
+
+    /**
+     * @brief Flags of useful headers found in the response.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    uint32_t flags;
```

Comment:
Can this just be an 8 (or 16) bit map of flags instead of 32 bits?

Reply:
Same question for the `HTTPRequestInfo_t.flags` member

Reply:
I want to leave this as 32 this add flexibility to add more later.

Reply:
I want to leave this as 32 this add flexibility to add more later.